launcher/README.md

```diff
@@ -0,0 +1,110 @@
+# Start your extension from the launcher
+
+In this example, you will learn how to start your extension from the launcher and how to have optional
+dependencies to JupyterLab features.
+
+![Launcher example](preview.gif)
+
+> Acknowledgement: This example is copied from Jeremy Tuloup [Python file extension](https://github.com/jtpio/jupyterlab-python-file).
+
+In this example, you will add the ability to create an empty Python file. To do so,
+your extension will use two commands defined by the [documents manager](https://github.com/jupyterlab/jupyterlab/blob/master/packages/docmanager-extension/src/index.ts#L47-L75) of JupyterLab:
+
+- `'docmanager:new-untitled'`: Create new untitled document
+- `'docmanager:open'`: Open a document
+
+The command will create a new Python file and then open it:
+
+```ts
+// src/index.ts#L36-L58
+
+commands.addCommand(CommandIDs.createNew, {
+  label: args => (args['isPalette'] ? 'New Python File' : 'Python File'),
+  caption: 'Create a new Python file',
+  iconClass: args => (args['isPalette'] ? '' : ICON_CLASS),
+  execute: async args => {
+    // Get the directory in which the Python file must be created;
+    // otherwise take the current filebrowser directory
+    let cwd = args['cwd'] || browserFactory.defaultBrowser.model.path;
+
+    // Create a new untitled python file
+    const model = await commands.execute('docmanager:new-untitled', {
+      path: cwd,
+      type: 'file',
+      ext: 'py'
+    });
+
+    // Open the newly created file with the 'Editor'
+    return commands.execute('docmanager:open', {
+      path: model.path,
+      factory: FACTORY
+    });
+  }
+});
+```
+
+To link that command to the JupyterLab launcher, the `ILauncher` interface needs to be passed to the `activate`
+extension function. As that interface is provided by the `@jupyterlab/launcher` package, it needs first to be installed:
+
+```bash
+jlpm add @jupyterlab/launcher
+```
+
+Then you can use it in the extension by importing it:
+
+```ts
+// src/index.ts#L10-L10
+
+import { ILauncher } from '@jupyterlab/launcher';
+```
+
+And finally you can request it as extension dependency:
+
+```ts
+// src/index.ts#L22-L33
+
+const extension: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab-examples/launcher',
+  autoStart: true,
+  optional: [ILauncher, IMainMenu, ICommandPalette],
+  requires: [IFileBrowserFactory],
```

Comment:
Should we position `requires` first, and then `optional`? So the order is the same as the parameters in the `activate` function.

Reply:
Not sure there is a clear pattern for that though (also in core lab).

Reply:
I agree with you. There is no clear pattern to have the attribute in alphabetical order. And as the required arguments come before the optional ones, it makes sense to list them in that order.